kernel/appenv.c

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <myst/args.h>
+#include <myst/eraise.h>
+#include <myst/file.h>
+#include <myst/fsgs.h>
+#include <myst/json.h>
+#include <myst/kernel.h>
+#include <myst/printf.h>
+
+#define APPENV_FILENAME "appenv.json"
+
+typedef struct json_callback_data
+{
+    myst_args_t env;
+} json_callback_data_t;
+
+static int _concat_env_variables(
+    const char* value,
+    myst_args_t* new_envp,
+    int pos)
+{
+    char* tuple;
+    const char* curr = new_envp->data[pos];
+
+    if (!curr)
+        return -EINVAL;
+
+    /* Append the APPENV_FILENAME value of the var to the config.json value */
```

Comment:
The comment about `APPENV_FILENAME` and `config.json` is too specific.

Reply:
I agree, modified it a little